spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'json'
+describe ManageIQ::Providers::Nuage::NetworkManager::VsdClient do
+  before(:each) do
+    @userid = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+    @password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+    @hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+
+    # Ensure that VCR will obfuscate the basic auth
+    VCR.configure do |c|
+      # workaround for escaping host
+      c.before_playback do |interaction|
+        interaction.filter!(CGI.escape(@hostname), @hostname)
+        interaction.filter!(CGI.escape('NUAGE_NETWORK_HOST'), 'nuagenetworkhost')
+      end
+      c.filter_sensitive_data('NUAGE_NETWORK_AUTHORIZATION') { Base64.encode64("#{@userid}:#{@password}").chomp }
+    end
+  end
+
+  it "should return valid non empty json responses" do
+    VCR.use_cassette('manageiq/providers/nuage/network_manager/vsd_client/vsd_client', :allow_unused_http_interactions => true, :record => :new_episodes) do
+      vsd_client = ManageIQ::Providers::Nuage::NetworkManager::VsdClient.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, @password)
+
+      enterprises_resp = vsd_client.get_enterprises
+      domains_resp = vsd_client.get_domains
+      zones_resp = vsd_client.get_zones
+      subnets_resp = vsd_client.get_subnets
+      vports_resp = vsd_client.get_vports
+      vms_resp = vsd_client.get_vms
+      policy_groups_resp = vsd_client.get_policy_groups
+
+      assert_object_not_empty(enterprises_resp)
```

Comment:
Are we sure, we want to always expect 2 elements in this array? I know that currently the `get_policy_groups` returns 2 objects in an array, but will it always be so? @miha-plesko @gberginc please comment on this.

Reply:
You are recording the actual call to the service. At the time this was recorded, there were exactly 2 which @miha-plesko suggests to test. However this does not imply that `vsd_client.get_enterprises` must always return `enterprises.count == 2`.

That's one of the reasons why we are recording the calls so that we are able to mostly work in controlled environment. We could mock the backend, true, but VCR is here to help us.

Reply:
As @gberginc said: the test and the VCR cassette can be bound together, I see no problem with that. Later when new VCR is recorded, the test must be updated accordingly.

Reply:
Ok, understood.